src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -128,5 +128,15 @@ private static void VerifyBytes(IntPtr p, int length)
                 Assert.Equal((byte)i, Marshal.ReadByte(p + i));
             }
         }
+
+        [Fact]
+        public static void GetHRForException()
+        {
+            Assert.Equal(0, Marshal.GetHRForException(null));
+
+            Exception e = new Exception();
+            Assert.True(Marshal.GetHRForException(e) < 0);
```

Comment:
Nit: Could be:
```
Assert.InRange(Marshal.GetHRForException(e), int.MinValue, -1);
```
The small advantage there is that the error message that'll be generated if this fails will include the details on the actual value and the range.